### PR TITLE
Minor: Improve `arrow_statistics` tests

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/datafusion/issues/10922

## Rationale for this change

I thought of a way to make the tests easier to write -- which will be important as we start expanding out the test coverage

## What changes are included in this PR?

Add an enum to make more explicit what is being testsed

## Are these changes tested?

Only tests

## Are there any user-facing changes?

No

cc @marvinlanhenke

### CODE REVIEW

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 1]
```diff
@@ -164,6 +164,36 @@ impl TestReader {
     }
 }
 
+/// Which statistics should we check?
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:167]: I think this is easier to understand (and now it is at least documented why sometimes the page stats are the same and some they are different)

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 2]
```diff
@@ -259,13 +288,16 @@ impl<'a> Test<'a> {
...
-            let expected_row_counts = Arc::new(expected_row_counts) as ArrayRef;
+            // https://github.com/apache/datafusion/issues/10926
+            let expected_row_counts: ArrayRef = Arc::new(expected_row_counts.clone());
             assert_eq!(
                 &row_counts, &expected_row_counts,
                 "{column_name}: Mismatch with expected row counts. \
                 Actual: {row_counts:?}. Expected: {expected_row_counts:?}"
             );
-        } else {
+        }
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:298]: The code can now check either or both types of statistics

File: datafusion/core/tests/parquet/arrow_statistics.rs [hunk 3]
```diff
@@ -482,22 +513,20 @@ async fn test_int_64() {
...
+        // mins are [-5, -4, 0, 5]
+        expected_min: Arc::new(Int64Array::from(vec![-5, -4, 0, 5])),
+        // maxes are [-1, 0, 4, 9]
+        expected_max: Arc::new(Int64Array::from(vec![-1, 0, 4, 9])),
+        // nulls are [0, 0, 0, 0]
+        expected_null_counts: UInt64Array::from(vec![0, 0, 0, 0]),
+        // row counts are [5, 5, 5, 5]
+        expected_row_counts: UInt64Array::from(vec![5, 5, 5, 5]),
+        column_name: "i64",
+        check: Check::Both,
```
Comment (alamb) [inline datafusion/core/tests/parquet/arrow_statistics.rs:527]: I thought `Both` was easier to understand here than the loop -- and will be a better pattern to follow to add supprot for other types

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb 
the enum Both probably will require modifications though if there is more Check variants in future

**Review by Weijun-H — APPROVED** [review]
LGTM! Thanks @alamb

**Review by marvinlanhenke — APPROVED** [review]
thanks lgtm

**Comment by alamb** [comment]
Thanks everyone for the reviews

**Pull request merged**